lib/sync-util.js

```diff
@@ -1,11 +1,27 @@
 var crypto = require('crypto');
 var winston = require('winston');
+var asyncJson = require('async-json');
 var moment = require('moment');
 var assert = require('assert');
 
 var SYNC_LOGGER = 'SYNC';
 var loggers = {};
 
+var asyncGenerateHash = function(plainText, cb) {
+  if(plainText) {
+    console.log('generateHash - async mode');
+    return asyncSortedStringify(plainText, function(err, hash) {
+      console.log('generateHash response - ' + hash);
+      var shasum = crypto.createHash('sha1');
+      shasum.update(hash);
+      hash = shasum.digest('hex');
+      return cb(null, hash);
```

Comment:
Could we perhaps just pass the new hash to the callback instead of modifying the hash passed to this function?
```javascript
return cb(null, shasum.digest('hex'));
```